pkg/epp/datastore/datastore.go

```diff
@@ -69,24 +68,25 @@ type Datastore interface {
 	Clear()
 }
 
-func NewDatastore(parentCtx context.Context, epFactory datalayer.EndpointFactory, modelServerMetricsPort int32) Datastore {
+func NewDatastore(parentCtx context.Context, epFactory datalayer.EndpointFactory, modelServerMetricsPort int32, endPointsPool *datalayer.EndPointsPool) Datastore {
 	store := &datastore{
-		parentCtx:              parentCtx,
-		poolAndObjectivesMu:    sync.RWMutex{},
-		objectives:             make(map[string]*v1alpha2.InferenceObjective),
-		pods:                   &sync.Map{},
-		modelServerMetricsPort: modelServerMetricsPort,
-		epf:                    epFactory,
+		parentCtx:                parentCtx,
+		endPointsAndObjectivesMu: sync.RWMutex{},
```

Comment:
no need to change everything to endpoints, pool was a good name already since the type is called `EndpointPool`